### PR TITLE
Extend concurrency property to accept percentage

### LINKED ISSUE #1652: Concurrency in percent
I think extending the API of the `concurrency`property in the `renderMedia()` function by a percentage value would be a great addition, for example `100%` using all CPU Threads avaibale on the machine.

### PR DESCRIPTION
**Pull request by JRavi2**
Closes #1652

### CODE REVIEW

File: packages/docs/docs/renderer/render-frames.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ A `string` which must be either `png`, `jpeg` or `none`.
 
 _optional_
 
-A `number` specifying how many render processes should be started in parallel or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
+A `number` specifying how many render processes should be started in parallel, a `string` specifying the percentage of the cpu threads to use, or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
```
Comment (akx) [inline packages/docs/docs/renderer/render-frames.md]:
```suggestion
A `number` specifying how many render processes should be started in parallel, a `string` specifying the percentage of the CPU threads to use, or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
```

File: packages/docs/docs/renderer/render-media.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ An object of arbitrary shape that will be passed as [props to your composition](
 
 _optional_
 
-A `number` specifying how many render processes should be started in parallel or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
+A `number` specifying how many render processes should be started in parallel, a `string` specifying the percentage of the cpu threads to use, or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
```
Comment (akx) [inline packages/docs/docs/renderer/render-media.md]:
```suggestion
A `number` specifying how many render processes should be started in parallel, a `string` specifying the percentage of the CPU threads to use, or `null` to let Remotion decide based on the CPU of the host machine. Default is half of the CPU threads available.
```

File: packages/renderer/src/validate-concurrency.ts [hunk 3]
```diff
@@ -7,23 +7,27 @@ export const validateConcurrency = (value: unknown, setting: string) => {
...
-		);
+		if (value > require('os').cpus().length) {
+			throw new Error(
+				`${setting} is set higher than the amount of CPU cores available. Available CPU cores: ${
+					require('os').cpus().length
+				}, value set: ${value}`
+			);
+		}
+	} else if(!(/^\d+(\.\d+)?%$/.test(value))) {
+		throw new Error(setting + ' must be a number or percentage, but is ' + value);
```
Comment (akx) [inline packages/renderer/src/validate-concurrency.ts]: This should probably use a template string too.

File: packages/renderer/src/validate-concurrency.ts [hunk 4]
```diff
@@ -7,23 +7,27 @@ export const validateConcurrency = (value: unknown, setting: string) => {
...
-			}, value set: ${value}`
-		);
+		if (value > require('os').cpus().length) {
+			throw new Error(
+				`${setting} is set higher than the amount of CPU cores available. Available CPU cores: ${
+					require('os').cpus().length
+				}, value set: ${value}`
+			);
+		}
+	} else if(!(/^\d+(\.\d+)?%$/.test(value))) {
```
Comment (akx) [inline packages/renderer/src/validate-concurrency.ts]: This allows a floating point number, but you're using `parseInt` elsewhere. Will that work?

**Review by akx — CHANGES_REQUESTED** [review]

File: packages/renderer/src/validate-concurrency.ts (hunk 4 above)
Comment (JonnyBurger) [reply packages/renderer/src/validate-concurrency.ts]: Seems like it will round down the number then. I'm fine with that behavior

**Comment by JonnyBurger** [comment]
And thanks for the contribution 🙌  will pay out the bounty today.

**Comment by JRavi2** [comment]
@JonnyBurger in `getActualConcurrency` the calculated value is [floored](https://github.com/remotion-dev/remotion/pull/1668/files#diff-499d509d3834565a806420e611c780d6e649a227ea458bf1ba22c59484baeb9fR16), hence the failing test? Should it be rounded instead?

**Comment by JonnyBurger** [comment]
@JRavi2 floor seems better! I changed the tests

**Comment by JRavi2** [comment]
Changed a test to rounded for the `null` case, since that was how it was being handled earlier

**Comment by JonnyBurger** [comment]
Awesome, thanks a lot!

**Pull request merged**